sd-array/task_1.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+using namespace std;
+
+int main(){
+    const int MAX = 100;
+    int n, a[MAX];
+    cin >> n;
+    for(int i = 0; i < n; i++){
+        cin >> a[i];
+    }
+    int start = 0, end = 0, max = 0, curr_start = 0, curr_end = 0, curr_max = 1;
+    for(int i = 0; i < n - 1; i++){
+        if(a[i] == a[i + 1]){
+            curr_max++;
+            curr_end = i + 1;
+        }else{
+            curr_start = i + 1;
+            curr_max = 1;
+            curr_end = curr_start;
+        }
+        if(curr_max > max){
+            max = curr_max;
+            start = curr_start;
+            end = curr_end;
+        }
+    }
```

Comment:
```suggestion
    for(int i = 0; i < n - 1; i++){
        if(a[i] == a[i + 1]){
            curr_max++;
        }else{
            curr_start = i + 1;
            curr_max = 1;
        }
        if(curr_max > max){
            max = curr_max;
            start = curr_start;
            end = i;
        }
    }
   if(curr_max > max){
       max = curr_max;
       start = curr_start;
       end = n-1;
   }
```
или

```suggestion
    a[n] = a[n-1] + 1;
    for(int i = 0; i < n - 1; i++){
        if(a[i] == a[i + 1]){
            curr_max++;
        }else{
            curr_start = i + 1;
            curr_max = 1;
        }
        if(curr_max > max){
            max = curr_max;
            start = curr_start;
            end = i;
        }
    }
```

Това са двата варианта да не изпуснеш решение (виж долния коментар)